### PR TITLE
Backport minor changes from #2656 (Custom Tags)

### PR DESCRIPTION
**Pull request by uklotzde**
A first set of minor changes backported from #2656. Please refer to the individual commit messages. Some extensions are not used yet, but extracting them beforehand helps to shrink the scope of #2656.

More PRs to follow. One after another.

### CODE REVIEW

**Comment by uklotzde** [comment]
I have deliberately decided not to base this PR on 2.3, even though it would be possible. Let's release 2.3 as is and restrict all future work to main.

File: src/engine/enginebuffer.cpp [hunk 1]
```diff
@@ -523,7 +523,7 @@ void EngineBuffer::loadFakeTrack(TrackPointer pTrack, bool bPlay) {
         m_playButton->set((double)bPlay);
     }
     slotTrackLoaded(pTrack, pTrack->getSampleRate(),
-                    pTrack->getSampleRate() * pTrack->getDurationInt());
+                    pTrack->getSampleRate() * pTrack->getDurationSecondsInt());
```
Comment (Holzhaus) [inline src/engine/enginebuffer.cpp:526]: Pre-commit doesn't like the unnecessary line break.

File: src/engine/enginebuffer.cpp [hunk 2]
```diff
@@ -522,8 +522,10 @@ void EngineBuffer::loadFakeTrack(TrackPointer pTrack, bool bPlay) {
     if (bPlay) {
         m_playButton->set((double)bPlay);
     }
-    slotTrackLoaded(pTrack, pTrack->getSampleRate(),
-                    pTrack->getSampleRate() * pTrack->getDurationInt());
+    slotTrackLoaded(
+            pTrack,
+            pTrack->getSampleRate(),
+            pTrack->getSampleRate() * pTrack->getDurationSecondsInt());
```
Comment (daschuer) [inline src/engine/enginebuffer.cpp:528]: No change by this PR, but why we need to round to seconds, when we multiply by samples anyway? 
I think rounding after multiplication would be correct.

Comment (uklotzde) [reply src/engine/enginebuffer.cpp:528]: Only used in tests. I don't wanto to modify any behavior.

Comment (Holzhaus) [reply src/engine/enginebuffer.cpp:528]: Can you add a TODO that we should investigate this?

**Comment by uklotzde** [comment]
It is annoying that all CI actions in forked repos are failing since recently:

https://github.com/uklotzde/mixxx/actions/runs/818369140

Is anyone aware of this?

**Comment by Holzhaus** [comment]
> It is annoying that all CI actions in forked repos are failing since recently:
> 
> https://github.com/uklotzde/mixxx/actions/runs/818369140
> 
> Is anyone aware of this?

I'll fix it. In thr meantime, just push some tags to your fork.

**Comment by daschuer** [comment]
I had also already a look, just building in: https://github.com/daschuer/mixxx/runs/2525042562

**Comment by daschuer** [comment]
https://github.com/daschuer/mixxx/runs/2525182098

**Review by daschuer — APPROVED** [review]

**Comment by daschuer** [comment]
Thank you.

**Pull request merged**